packages/oft-sand/contracts/oft/OFTAdapter.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IERC20Metadata, IERC20} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {OFTCore} from "./OFTCore.sol";
+
+/**
+ * @title OFTAdapter Contract
+ * @dev OFTAdapter is a contract that adapts an ERC-20 token to the OFT functionality.
```

Comment:
Add a dev note that says this is based on @layerzerolabs/lz-evm-oapp-v2/contracts/oft/OFTAdapter.sol but modified to allow for ERC2771 compatibility

Reply:
(Also do this for any other relevant contracts where we've made modifications)